File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if(nums.size()<3) return false;
        int fmin=INT_MAX,smin=INT_MAX;
        for(int& num :nums){
            if(num<=fmin) fmin=num;
            else if(num<=smin) smin=num;
            else return true;
        }
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first,second=inf,inf
        for ele in nums:
            if ele<first:
                first=ele
            elif ele<second and ele!=first:
                second=ele
            if ele>first and ele>second:
                return 1
        return 0